In [1]:
import pandas as pd
import numpy as np

## Dataset Import

In [2]:
primary_address = '\\Users\dsm3\OneDrive - UFPE\Documentos\Data Science\Residência CIN (2021)\Disciplinas\TCC\TCC - Datasets\\tweets-rotulados\\'
secondary_address = 'tweets-220826-220707-200-class_-1-labeled-zeta_3.csv'
PATH = primary_address + secondary_address

df_neg = pd.read_csv(PATH, usecols=['index', 'date', 'tweet', 'tweet_clean', 'emojis', 'label', 'label_check', 'file'], infer_datetime_format=True, parse_dates=[3])

secondary_address = 'tweets-220826-220707-200-class_0-labeled-zeta_3.csv'
PATH = primary_address + secondary_address

df_neutral = pd.read_csv(PATH, usecols=['index', 'date', 'tweet', 'tweet_clean', 'emojis', 'label', 'label_check', 'file'], infer_datetime_format=True, parse_dates=[3])

secondary_address = 'tweets-220826-220707-200-class_1-labeled-zeta_3.csv'
PATH = primary_address + secondary_address

df_pos = pd.read_csv(PATH, usecols=['index', 'date', 'tweet', 'tweet_clean', 'label', 'emojis', 'label_check', 'file'], infer_datetime_format=True, parse_dates=[3])

print('Shape - Negative class: ', df_neg.shape)
print('Shape - Neutral class: ', df_neutral.shape)
print('Shape - Positive class: ', df_pos.shape)

Shape - Negative class:  (200, 8)
Shape - Neutral class:  (200, 8)
Shape - Positive class:  (200, 8)


In [3]:
df_neg.head()

index                       date  \
0  288189  2022-08-25 23:59:45+00:00   
1  216611  2022-08-26 03:00:05+00:00   
2  130145  2022-08-03 00:12:57+00:00   
3  288351  2022-08-25 23:59:38+00:00   
4   59722  2022-07-14 10:43:49+00:00   

                                               tweet  \
0         Esse Lula me dá enjoo 🤢… ladrão, picareta.   
1                  Lula 🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮   
2  Eleições no Grêmio nesse ano e vocês aí preocu...   
3  Nunca assistam uma entrevista do lula com uma ...   
4  Véi me dá um ódio quando vem alguém no posto e...   

                                         tweet_clean emojis  \
0          Esse Lula me dá enjoo … ladrão, picareta.      🤢   
1                                              Lula       🤮   
2  Eleições no Grêmio nesse ano e vocês aí preocu...      🤦   
3  Nunca assistam uma entrevista do lula com uma ...      😡   
4  Véi me dá um ódio quando vem alguém no posto e...      🤬   

                                           file  label  label_check  
0  lula-220826-220825-84492-(entrevista_JN).csv   -1.0           -1  
1  lula-220826-220825-84492-(entrevista_JN).csv   -1.0           -1  
2                  lula-220804-220728-26269.csv   -1.0            0  
3  lula-220826-220825-84492-(entrevista_JN).csv   -1.0            0  
4                  lula-220714-220707-22495.csv   -1.0           -1

## Analysis

In [4]:
100 * df_neg['label_check'].value_counts()/200

-1    54.5
 0    42.0
 1     3.5
Name: label_check, dtype: float64

In [5]:
100 * df_neutral['label_check'].value_counts()/200

 0.0    78.0
 1.0    13.5
-1.0     8.0
Name: label_check, dtype: float64

In [6]:
100 * df_pos['label_check'].value_counts()/200

 1    68.0
 0    24.5
-1     7.5
Name: label_check, dtype: float64

* Em uma amostra aleatória de 200 registros de cada classe rotulada a partir dos emojis, tem-se:
1. **Classe Negativa:** 
    1. 54.5% dos registros foram corretamente rotulados;
    2. Dos registros incorretamente rotulados, 42% pertenciam a classe neutra e 3.5% pertenciam a classe positiva.
<br>
2. **Classe Neutra:**
    1. 78.0% dos registros foram corretamente rotulados;
    2. Dos registros incorretamente rotulados, 13.5% pertenciam a classe negativa e 8.0% pertenciam a classe positiva.
<br>
3. **Classe Positiva:**
    1. 68.0% dos registros foram corretamente rotulados;
    2. Dos registros incorretamente rotulados, 24.5% pertenciam a classe neutra e 7.5% pertenciam a classe negativa.
<br>